src/compiler/checker.ts

```diff
@@ -22721,6 +22718,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         const filtered = contains(include, Ternary.False) ? getUnionType(types.filter((_, i) => include[i])) : target;
         return filtered.flags & TypeFlags.Never ? target : filtered;
+
+        function getTypeOfPropertyOrIndexSignatureOfType(type: Type, name: __String): Type | undefined {

```

Comment:
It might be less confusing to have this next to getTypeOfPropertyOrIndexSignatureOfType.
Well, initially more confusing for the reader, because they'll have to ask "Why are there *two* here?". But they'll be *aware* that there are two nearly identical things that need to be used in different situations.